app/agents/voice/automatic/tools/juspay/analytics.py

```
@@ -712,80 +712,379 @@ async def delete_euler_offer(params: FunctionCallParams):
+            "offer_description": {
+                "title": offer_title or "Updated Offer",
+                "description": offer_description or "Updated offer description",
+                "tnc": "",
+                "sponsored_by": sponsored_by or "BREEZE",
+                "display_title": offer_title or "Updated Offer"
+            },
+            "ui_configs": {
+                "is_hidden": "false",
+                "should_validate": "true",
+                "auto_apply": "true",
+                "offer_display_priority": 0,
+                "payment_method_label": ""
+            },
+            "rule_dsl": {
+                "order": {
+                    "max_quantity": None,
+                    "min_quantity": None,
+                    "max_order_amount": None,
+                    "min_order_amount": str(min_order_amount or 1),
+                    "currency": "INR",
+                    "amount_info": []
+                },
+                "additional_payment_filters": None,
+                "payment_instrument": payment_instruments_payload or list(instrument_map.values()),
+                "counters": [],
+                "txn_type": ["ORDER"],
+                "payment_channel": [],
+                "benefits": [
+                    {
+                        "type": offer_type or "DISCOUNT",
+                        "calculation_rule": calculation_type or "ABSOLUTE",
+                        "value": discount_value or 10,
+                        "amount_info": [],
+                        "max_amount": max_discount_amount,
+                        "global_max_amount": None
+                    }
+                ],
+                "filters": {
+                    "blacklist": [],
+                    "whitelist": []
+                }
+            },
+            "status": status or "ACTIVE",
+            "start_time": start_date_iso or "2025-08-25T18:30:00.000Z",
+            "end_time": end_date_iso or "2025-09-26T18:29:59.000Z",
```

If `start_date_iso` or `end_date_iso` is None, why not fail the operation instead of defaulting to arbitrary dates?

---

Now fetching all the details from the existing offer details, removed this fallback